osh/cmd_parse.py

```diff
@@ -1422,31 +1427,38 @@ def ParseCase(self):
     case_node.to_match = to_match
     self._Next()
 
+    in_kw = None  # type: Token
+
     self._NewlineOk()
-    in_spid = word_.LeftMostSpanForWord(self.cur_word)
     self._Peek()
     if self.parse_opts.parse_brace() and self.c_id == Id.Lit_LBrace:
       self._Next()
     else:
       self._Eat(Id.KW_In)
+      in_kw = word_.AsKeywordToken(self.cur_word)
     self._NewlineOk()
 
     if self.c_id != Id.KW_Esac:  # empty case list
-      self.ParseCaseList(case_node.arms)
+      esac_kw = self.ParseCaseList(case_node.arms)
       # TODO: should it return a list of nodes, and extend?
       self._Peek()
+    else:
+        esac_kw = word_.AsKeywordToken(self.cur_word)
```

Comment:
indentation looks messed up here

(We have an open issue to autoformat the code)